### DISCUSSION TITLE
Sieve_of_Eratosthenes in V

### DISCUSSION THREAD

**Discussion by config2000** [vlang/v#21867] [Questions and Answers]
Hello coders.
I am currently enjoying playing around with V. What do you guys think of this first translation example of **Sieve_of_Eratosthenes**?

Perhaps with some timing routing it could be useful to compare V's speed against other languages.

```
// https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes

fn sieve(n int) []int {
	mut primes:= []bool{len: n+1, init: true}

	for p:= 2; p*p <= n; p++ {
		if primes[p] == true {
		    for i:= p*2; i<=n; i+=p { primes[i] = false}
		}
	}

	mut prime_numbers:= []int{}
	for p:=2; p<=n; p++ {
	    if primes[p] {prime_numbers << p}
	}
	
	return prime_numbers
}

fn main() {
	println(sieve(25))
}
```

**Comment by larpon** [comment]
I think a PR contributing it to the `examples` folder would be welcomed. Using it as a benchmark may or may not be a good idea. Benchmarks tend to be unreliable and hard to reproduce across a lot of different hardware for a number of reasons

**Reply by config2000** [reply]
I agree. The examples are in fact a good learning source. "Learn by example" is a very good way to understand a programming language.

**Comment by Wajinn** [comment]
There is a [Sieve example](https://rosettacode.org/wiki/Sieve_of_Eratosthenes#V_(Vlang)) on Rosetta Code, to compare with or add to.  Additionally, Exercism has [Sieve](https://exercism.org/tracks/vlang/exercises/sieve) as well. This or any other solution could be added.

**Reply by config2000** [reply]
Very nice. There are quite a lot of **v** examples there. Another good learning resource.
I have added a Factorial Primes v (vlang) example that was not present ([link](https://rosettacode.org/wiki/Factorial_primes#V_(Vlang)))

**Reply by Wajinn** [reply]
Adding that example was great. In fact, V's Rosetta Code is so close to 500 examples and Exercism is so close to 100.
